test/object-store/util/sync/redirect_server.hpp

```diff
@@ -264,16 +351,64 @@ class RedirectingHttpServer {
                     return;
                 }
 
-                send_simple_response(conn, HTTPStatus::NotFound, "Not found", {});
+                // Send redirect response for appservices calls
+                // Starts with 'http' and contains api path
+                if (req.path.find("/api/client/v2.0/") == 0) {
+                    use_301 = !use_301;
+                    auto status = use_301 ? HTTPStatus::MovedPermanently : HTTPStatus::PermanentRedirect;
+                    auto reason = use_301 ? "Moved Permanently" : "Permanent Redirect";
+                    auto location = m_redirect_to_base_url + req.path;
+                    auto redirect = util::format("<html><body><p>%1 to <a href=\"%2\">%2</a></p></body></html>",
```

Comment:
why do we need a body for this? is this the body that the real server sends or something?